tt_metal/llrt/blackhole/bh_hal_idle_eth.cpp

```diff
@@ -33,8 +35,8 @@ HalCoreInfoType create_idle_eth_mem_map() {
     mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::WATCHER)] = GET_IERISC_MAILBOX_ADDRESS_HOST(watcher);
     mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::DPRINT)] = GET_IERISC_MAILBOX_ADDRESS_HOST(dprint_buf);
     mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::PROFILER)] = GET_IERISC_MAILBOX_ADDRESS_HOST(profiler);
-    mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::KERNEL_CONFIG)] = IDLE_ERISC_L1_KERNEL_CONFIG_BASE;
-    mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::UNRESERVED)] = ((L1_KERNEL_CONFIG_BASE + L1_KERNEL_CONFIG_SIZE - 1) | (max_alignment - 1)) + 1; // TODO: this is wrong, need idle eth specific value
+    mem_map_bases[utils::underlying_type<HalL1MemAddrType>(HalL1MemAddrType::KERNEL_CONFIG)] = MEM_IERISC_MAP_END;
```

Comment:
eventually (now?): the HAL ends at UNRESERVED, the KERNEL_CONFIG lives above that.  the allocator (or device?) doles that memory out.  at that point, KERNEL_CONFIG is just a buffer and the device side knows nothing about it (just has pointers to memory)

Reply:
yeah this is a todo on me, to avoid merge conflicts with myself I can add a commit for this in this pr

Reply:
Wilder made some changes related to kernel config, I can sync with him to uplift his PR to remove it from the HAL and merge this as is